src/shell/src/command/list.rs

```diff
@@ -66,10 +70,10 @@ impl Command<InitializedListCommand> {
                        "Unknown  "
                    },
                     child.name().unwrap().to_string_lossy()
-                );
+                )?;
```

Comment:
What is the purpose of the question mark on this line?

Reply:
Given T as a raw value result type and E as an error type, the std "maybe" ( monadic ) object `Result` can contain a `Ok(T)` or `Err(E)`, 

`?` operator is a kind of shortcut for following behavior : "If `Result` is `Err` then return the `Err` else if `Ok` unwrap result"

It's a way to replace needs around throw / try / catch, but in a more functional way.

This operator wouldn't be usable if the wrapping function hadn't return type as ` -> Result<(), CommandError> ` ( `()` is void type )

Currently it works only over `Result` or `Option` objects but future rust will support customizing this operator .